### PR TITLE
litestar 2.12.1 :snowflake:

### PR DESCRIPTION
**Pull request by psteyer**
litestar 2.12.1 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5394]
- dev_url:        https://github.com/litestar-org/litestar/tree/v2.12.1
- conda_forge:    https://github.com/conda-forge/litestar-feedstock/blob/main/recipe/meta.yaml
- pypi:           https://pypi.org/project/litestar/2.12.1
- pypi inspector: https://inspector.pypi.io/project/litestar/2.12.1

### Explanation of changes:

- new version number

[PKG-5394]: https://anaconda.atlassian.net/browse/PKG-5394?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lorepirri** [comment]
@boldorider4 could you please restore and comment out or skip the packages that cannot be built because of missing dependencies? In future versions we might be able to enable them. We discussed this with @psteyer and in the ticket are all the notes

**Comment by boldorider4** [comment]
> @boldorider4 could you please restore and comment out or skip the packages that cannot be built because of missing dependencies? In future versions we might be able to enable them. We discussed this with @psteyer and in the ticket are all the notes

Fair enough. I was not a big fan of having large portions of code commented out so I went with a cleaner commented section, since I figured we would need to update the pinning anyway in future builds, but I can just revert the last commit.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,20 +32,19 @@ outputs:
...
         - hatchling
-        - python >=3.8
+        - python
       run:
-        - annotated-types
-        - python >=3.8
+        - python
         - anyio >=3
-        - click
-        - exceptiongroup
```
Comment (danpetry) [inline recipe/meta.yaml:44]: exceptiongroup, importlib-metadata, importlib-resources seem to be needed according to upstream

File: recipe/meta.yaml [hunk 2]
```diff
@@ -133,40 +138,44 @@ outputs:
...
-      requires:
-        - pip
-    about:
-      home: https://litestar.dev
-      dev_url: https://github.com/litestar-org/litestar
-      doc_url: https://docs.litestar.dev
-      summary: Light-weight and flexible ASGI API Framework (with cli)
-      license: MIT
-      license_file: LICENSE
+  # - name: litestar-with-cli
```
Comment (danpetry) [inline recipe/meta.yaml:146]: neater to just delete I think

Comment (danpetry) [reply recipe/meta.yaml:146]: oh this has been discussed already. fine

**Review by danpetry — APPROVED** [review]
looks fine; I'm not sure about the whole litestar-with stuff (just put it all in one package?) but I suppose this has already been discussed/decided

**Review by Andrii-Osipov — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (boldorider4) [reply recipe/meta.yaml:44]: good catch on!

**Pull request merged**